build64/debug_memtest.sh

```diff
@@ -0,0 +1,201 @@
+#! /bin/bash
+
+#########################
+##
+##	TODO: DESCRIPTION
+##
+
+Info() {
+	echo "Error: Invalid option"
+	echo "Type $0 -h for more information"
+}
+
+Help() {
+	echo "TODO Add description..."
+	echo
+	echo "Syntax: $0 [-h|c|t <terminal><execution_command>]"
+	echo "options:"
+	echo "h	print this help"
+	echo "c	delete all debugging related files"
+	echo "t define alternative terminal"
+	echo
+}
+
+Terminal_Help() {
+	echo "You can define your own terminal inclusive its execution command via:"
+	echo "./debug_script.sh -t \"<terminal> <execution_command> \""
+	echo "See following examples:"
+	echo "./debug_script.sh -t \"x-terminal-emulator -e \""
+	echo "./debug_script.sh -t \"gnome-terminal --  \""
+    echo ".debug_script.sh -t \"xterm -e \""
+}
+
+Clear() {
+    echo "Deleting files..."
+    rm -rf hda-contents
+    rm -f debug.log
+    rm -f gdbscript
+    rm -f QemuKill
+    make clean
+    rm -f OVMF.fd
+    rm -f OVMF_VARS.fd
+    rm -f OVMF_CODE.fd
+    rm -f memtest_shared_debug.lds
+}
+
+while getopts ":hct:" option; do
+
+	case $option in
+		h) # display Help
+			Help
+			exit;;
+		c) # clear directory
+			Clear
+			exit;;
+		t) # define own terminal
+			echo "argument -t called with parameter $OPTARG" >&2
+            if [ $OPTARG = "help" ]; then
+                Terminal_Help
+                exit 0
+            fi
+			TERMINAL="$OPTARG"
+			if ! $TERMINAL ls; then
+				echo "Your entered command is not valid. Please check it again"
+                echo "Or type \"./debug_memtest.sh -t help\" for help"
+				exit 1
+			fi
+			exit;;
+        \?) # invalid option
+            Info
+            exit;;
+	esac
+
+done
+
+Check() {
+	# Check if QEMU and OVMF is installed
+	if ! dpkg -l qemu-system* > /dev/null 2>&1; then
```

Comment:
Wouldn't a check based on `command -v` be enough here ? That would be much more portable.

Reply:
Yes, that's right. I have added this some time before the other checks. Now all checks are consistent